contracts/oracle/MarketGovernanceOracle.sol

```diff
@@ -0,0 +1,338 @@
+        /// set actual rate
+        uint256 newChangeRate = calculateRate(
+            PCVOracle(pcvOracle).getLiquidVenuePercentage()
+        );
+        uint256 previousActualRate = actualChangeRate;
+
+        /// sanity check that new rate is lte max change rate
+        require(
+            newChangeRate <= maximumChangeRate,
+            "MGO: new change rate above max"
+        );
+        actualChangeRate = newChangeRate;
+
+        emit ActualRateUpdated(previousActualRate, newChangeRate);
+    }
+
+    /// @notice function to set the reference to the PCV oracle
+    /// callable only by governor
+    /// @param _pcvOracle address of the new pcv oracle
+    function setPcvOracle(address _pcvOracle) external onlyGovernor {
+        address oldOracle = pcvOracle;
+        pcvOracle = _pcvOracle;
+
+        emit PCVOracleUpdated(oldOracle, _pcvOracle);
+    }
+
+    /// ------------- Oracle Only API -------------
+
+    /// @notice only callable by the pcv oracle, updates the actual rate
+    /// @param liquidPercentage the percentage of PCV that is liquid
+    function updateActualRate(uint256 liquidPercentage) external onlyPCVOracle {
+        /// if liquidity is fine, no-op
+        if (liquidPercentage >= liquidityJumpTarget) {
+            return;
+        }
+        /// first compound interest
+        _compoundInterest();
+
+        /// if too illiquid, adjust rate up
+        /// find amount off target, figure out how much actual rate should be
+        /// set actual rate
+        uint256 newChangeRate = calculateRate(liquidPercentage);
+        uint256 previousActualRate = actualChangeRate;
+
+        /// sanity check that new rate is lte max change rate
+        require(
+            newChangeRate <= maximumChangeRate,
+            "MGO: new change rate above max"
+        );
+        actualChangeRate = newChangeRate;
+
+        emit ActualRateUpdated(previousActualRate, newChangeRate);
+    }
+
+    /// ------------- Helper Methods -------------
+
+    /// todo verify abs is fine here
+    /// Linear Interpolation Formula
+    /// (y) = y1 + abs(x − x1) * (abs(y2 − y1) / abs(x2 − x1))
+    /// @notice calculate linear interpolation and return ending price
+    /// @param x is time value to calculate interpolation on
+    /// @param x1 is starting time to calculate interpolation from
+    /// @param x2 is ending time to calculate interpolation to
+    /// @param y1 is starting price to calculate interpolation from
+    /// @param y2 is ending price to calculate interpolation to
+    function _calculateLinearInterpolation(
+        uint256 x,
+        uint256 x1,
+        uint256 x2,
+        uint256 y1,
+        uint256 y2
+    ) internal pure returns (uint256 y) {
+        uint256 firstDeltaX = x1 > x ? x1 - x : x - x1;
+        uint256 secondDeltaX = x2 > x1 ? x2 - x1 : x1 - x2;
+        uint256 deltaY = y2 > y1 ? y2 - y1 : y1 - y2;
+        uint256 product = (firstDeltaX * deltaY) / secondDeltaX;
```

Comment:
this only calculates absolute values, we should ensure this version of the formula works.